File: app/src/main/java/com/jon/cotgenerator/enums/DataFormat.java
package com.jon.cotgenerator.enums;

import android.content.SharedPreferences;

import com.jon.cotgenerator.utils.Key;

public enum DataFormat {
    XML("XML"),
    PROTOBUF("Protobuf");

    private final String protocol;

    DataFormat(String protocol) {
        this.protocol = protocol;
    }

    public String get() {
        return protocol;
    }

    public static DataFormat fromPrefs(SharedPreferences prefs) {
        return fromString(prefs.getString(Key.DATA_FORMAT, ""));
    }

    public static DataFormat fromString(String str) {
        switch (str) {
            case "XML": return XML;
            case "Protobuf": return PROTOBUF;
            default: throw new IllegalArgumentException("Unknown data format: " + str);
        }
    }
}

File: common/src/main/java/com/jonapoul/common/utils/DataFormat.kt
package com.jonapoul.common.utils

import android.content.SharedPreferences
import com.jonapoul.common.prefs.CommonPrefs
import com.jonapoul.sharedprefs.getStringFromPair

enum class DataFormat(private val string: String) {
    XML("XML"),
    PROTOBUF("Protobuf");

    override fun toString(): String = string

    companion object {
        fun fromPrefs(prefs: SharedPreferences): DataFormat {
            return fromString(
                prefs.getStringFromPair(CommonPrefs.DATA_FORMAT)
            )
        }

        fun fromString(formatString: String): DataFormat {
            return values().first { it.toString() == formatString }
        }
    }
}
